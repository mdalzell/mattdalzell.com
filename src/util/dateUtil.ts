export class DateUtil {
  public static getCurrentYear = (): number => {
    return new Date().getFullYear()
  }
}
