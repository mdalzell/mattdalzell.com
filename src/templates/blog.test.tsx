import React from 'react'
import renderer from 'react-test-renderer'

import { DateUtil } from '../util'
import BlogPage from './blog'

beforeEach(() => {
  DateUtil.getCurrentYear = jest.fn(() => 2019)
})

describe('BlogPage', () => {
  const mockProps = {
    data: {
      allMarkdownRemark: {
        edges: [
          {
            node: {
              fields: { slug: 'test-1' },
              frontmatter: {
                date: new Date('5/1/2019').toDateString(),
                tags: ['red', 'blue'],
                title: 'Test #1',
              },
              id: 1,
            },
          },
          {
            node: {
              fields: { slug: '/test-2' },
              frontmatter: {
                date: new Date('4/1/2019').toDateString(),
                tags: ['red'],
                title: 'Test #2',
              },
              id: 2,
            },
          },
        ],
        group: [
          { fieldValue: 'red', totalCount: 2 },
          { fieldValue: 'blue', totalCount: 1 },
        ],
      },
      featuredPosts: {
        edges: [],
      },
      markdownRemark: {
        html:
          'These blog posts are my own ill-conceived musings as a software engineer, and do not represent the views of any employer or client, past or present.',
      },
    },
    location: {
      search: '',
    },
  }
  it('should match snapshot for default view', () => {
    const component = renderer.create(<BlogPage {...mockProps} />).toJSON()

    expect(component).toMatchSnapshot()
  })
})
