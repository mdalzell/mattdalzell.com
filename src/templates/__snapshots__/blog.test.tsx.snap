// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPage should match snapshot for default view 1`] = `
Array [
  <div
    className="container"
  >
    <div
      className="hidden-sm"
    >
      <ul
        className="nav nav-list"
      >
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              <span
                className="glyphicon glyphicon-list-alt"
              />
              Resume
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://github.com/mdalzell"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              <span
                className="glyphicon glyphicon-wrench"
              />
              GitHub
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://stackoverflow.com/story/mattdalzell"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span>
              <span
                className="glyphicon glyphicon-console"
              />
              Stack Overflow
            </span>
          </a>
        </li>
        <li>
          <a
            href="mailto:contact@mattdalzell.com"
          >
            Email
          </a>
        </li>
      </ul>
    </div>
    <div>
      <div>
        <h1>
          Blog
        </h1>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "These blog posts are my own ill-conceived musings as a software engineer, and do not represent the views of any employer or client, past or present.",
            }
          }
        />
        <h3>
          Recent Posts
        </h3>
        <ul
          className="no-list-style"
        >
          <li>
            <a
              href="/blogtest-1"
            >
              Test #1 - Wed May 01 2019
            </a>
          </li>
          <li>
            <a
              href="/blog/test-2"
            >
              Test #2 - Mon Apr 01 2019
            </a>
          </li>
        </ul>
        <p>
          <a
            href="/blog/all"
          >
            View all posts &gt;&gt;&gt;
          </a>
        </p>
        <h3>
          Tags
        </h3>
        <ul
          className="tags"
        >
          <li>
            <a
              href="/blog/all?tag=red"
            >
              #red
            </a>
          </li>
          <li>
            <a
              href="/blog/all?tag=blue"
            >
              #blue
            </a>
          </li>
        </ul>
        <a
          href="/"
        >
          Return Home
        </a>
      </div>
    </div>
  </div>,
  <footer>
    <ul
      className="nav nav-list"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/blog"
        >
          Blog
        </a>
      </li>
      <li>
        <a
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            <span
              className="glyphicon glyphicon-list-alt"
            />
            Resume
          </span>
        </a>
      </li>
      <li>
        <a
          href="https://github.com/mdalzell"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            <span
              className="glyphicon glyphicon-wrench"
            />
            GitHub
          </span>
        </a>
      </li>
      <li>
        <a
          href="https://stackoverflow.com/story/mattdalzell"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            <span
              className="glyphicon glyphicon-console"
            />
            Stack Overflow
          </span>
        </a>
      </li>
      <li>
        <a
          href="mailto:contact@mattdalzell.com"
        >
          Email
        </a>
      </li>
    </ul>
    <div>
      Â© 
      2019
       Matt Dalzell
    </div>
  </footer>,
]
`;
